testcases/OpTestLparFreq.py

```diff
@@ -0,0 +1,217 @@
+        self.powersave_check_list = ['Static', 'Power Saving', 'Maximum Performance']
+        for i in range(3):
+            self.set_powersave_mode(self.powersave_modes_list[i])
+            time.sleep(5)
+            self.check_powersave_mode(self.powersave_check_list[i])
+
+
+    def ModeFreq_test(self):
+
+        # This test sets Powersave mode to different available modes and
+        # checks whether the frequency chages accordingly.
+        # Verifies if Average frequency matches with Platform reported frequency.
+        # Verifies if frequency is maximum in maximum Performance mode and least in powersave mode.
+
+        powersave_freq_list = []
+        for i in range(3):
+            self.set_powersave_mode(self.powersave_modes_list[i])
+            time.sleep(5)
+            powersave_freq_list.append(self.get_cpu_freq())
+        for i in range(3):
+            log.info("Average Frequency in %s mode = %s" %
+                     (self.powersave_modes_list[i], powersave_freq_list[i]))
+        powersave_plat_freq_list = []
+        output = self.console.run_command("ppc64_cpu --frequency")
+        for i in range(3, 6):
+            powersave_plat_freq_list.append(output[i].split()[2])
+        log.info("Platform reported frequency for [ powersave, max, static ] = %s" %
+                 powersave_plat_freq_list)
+        j = [2, 0, 1]
+        for i in range(3):
+            if round(float(powersave_freq_list[i]), 1) != round(float(powersave_plat_freq_list[j[i]]), 1):
+                log.info("Static mode frequency test failed")
+                self.failed_test.append("%s mode frequency test failed" % self.powersave_modes_list[i])
+            else:
+                log.info("Average frequency matches with Platform reported frequency for %s mode" %
+                         self.powersave_modes_list[i])
+        if powersave_freq_list[0] < powersave_freq_list[1]:
+            log.info("Static mode frequency is less than Powersave mode frequency")
+            self.failed_test.append("Frequency is not least with Powersave mode.")
+        if powersave_freq_list[2] < powersave_freq_list[0]:
+            log.info("Maximum Performance mode frequency is less than Static mode frequency.")
+            self.failed_test.append("Frequency is not maximum with maximum Performance mode.")
+
+
+    def Lpar_shutdown_freq_test(self):
+
+        # This test Set Powersave mode to "Staic Power Saver" mode when lpar is deactivated
+        # and check if the frequency is set properly when the lpar boots up
+
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o shutdown -n %s" %
+                                (self.system_name, self.lpar_name))
+        time.sleep(60)
+        self.set_powersave_mode(self.powersave_modes_list[0])
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o on -b norm -n %s" %
+                                (self.system_name, self.lpar_name))
+        time.sleep(120)
```

Comment:
ah why this sleep very is this also system depended in that case it should be user driven